### PR TITLE
feat: [CDS-2898]: Supporting SSH inherit for Source Module in Terraform

### PR DESCRIPTION
**Pull request by namantalaycha10**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)

### CODE REVIEW

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 7 of 13 files at r1, 10 of 10 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformStepHelper.java, line 484 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B2MaI7fNXSZ_HsEcv:-N0B2MaI7fNXSZ_HsEcw:bgoxxsw) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformStepHelper.java#L484)):*
> ```Java
>                 .toGitStoreConfigDTO());
>         builder.useConnectorCredentials(
>             configuration.getSpec().getConfigFiles().getModuleSource().isUseConnectorCredentials());
> ```

Let's check here if module source is not null first
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 809 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B4-Zj5K4GpvuNiacb:-N0B4-Zj5K4GpvuNiacc:b3xg1fa) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L809)):*
> ```Java
>   }
> 
>   public void exportingSSHKey(TerraformTaskNGParameters taskParameters, GitStoreDelegateConfig conFileFileGitStore)
> ```

nit: name should be exportSSHKey
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 812 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B3hVm-JQ-vpITCPnS:-N0B3hVm-JQ-vpITCPnT:b-rvo3yv) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L812)):*
> ```Java
>       throws IOException {
>     GitConfigDTO gitConfigDTO = (GitConfigDTO) conFileFileGitStore.getGitConfigDTO();
>     if (gitConfigDTO.getGitAuthType() == SSH) {
> ```

as discussed let's add some warnings in executions logs if user is using this option with non-ssh or not supported flow. Ideally we should have another methods somethings like configureCredentialsForModuleSource and then in that method add warnings and use exportSSHKey
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 818 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B384M01a4z8VEHzQC:-N0B384NDxfdV4Qd9t0x:b1lgrfd) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L818)):*
> ```Java
>         String sshKey = String.valueOf(sshSessionConfig.getKey());
>         String workingDir = getBaseDir(taskParameters.getEntityId());
>         String baseDir = Paths.get(Paths.get(System.getProperty(USER_DIR_KEY)).toString(), workingDir).toString();
> ```

I'm confused here why do we need to append USER_DIR_KEY? you can get absolute path by using Path.getAbsolutePath()
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 819 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B2eI31F6GsNwOFlWz:-N0B2eI31F6GsNwOFlX-:b-z47tby) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L819)):*
> ```Java
>         String workingDir = getBaseDir(taskParameters.getEntityId());
>         String baseDir = Paths.get(Paths.get(System.getProperty(USER_DIR_KEY)).toString(), workingDir).toString();
>         Files.createDirectories(Paths.get(baseDir, "sshKeyDir"));
> ```

nit: this should be const and name can be ".ssh" (more known to users)
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 820 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B2oKJ5iUIztx-fbln:-N0B2oKJ5iUIztx-fblo:b-ubfg7d) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L820)):*
> ```Java
>         String baseDir = Paths.get(Paths.get(System.getProperty(USER_DIR_KEY)).toString(), workingDir).toString();
>         Files.createDirectories(Paths.get(baseDir, "sshKeyDir"));
>         sshKeyPath = Paths.get(baseDir, "sshKeyDir", "sshKey").toString();
> ```

nit: sshKey should be const, and we can name it something like ssh.key
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 829 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B3Rzp4lfIaRbF_DHR:-N0B3Rzp4lfIaRbF_DHS:b-nyeujn) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L829)):*
> ```Java
>       File file = new File(Paths.get(sshKeyPath).toString());
> 
>       file.setWritable(false, false);
> ```

Let's add a comment here why we need this manipulation with file permission
___
*[125-cd-nextgen/src/main/java/io/harness/cdng/manifest/yaml/storeConfig/moduleSource/ModuleSource.java, line 29 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B28BBASFnz0BXQhQu:-N0B28BBASFnz0BXQhQv:b-3gcsna) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/125-cd-nextgen/src/main/java/io/harness/cdng/manifest/yaml/storeConfig/moduleSource/ModuleSource.java#L29)):*
> ```Java
> @RecasterAlias("io.harness.cdng.manifest.yaml.storeConfig.moduleSource.ModuleSource")
> public class ModuleSource {
>   @NotNull @Wither private boolean useConnectorCredentials;
> ```

as discussed this should be runtime input

**Review by namantalaycha10 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)* status: 27 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformStepHelper.java, line 484 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B2MaI7fNXSZ_HsEcv:-N0FjksY7Beynkm5YrPy:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformStepHelper.java#L484)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Let's check here if module source is not null first
</blockquote></details>

Done.
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 812 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B3hVm-JQ-vpITCPnS:-N0FjpFrEpmiHvwiVS2o:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L812)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

as discussed let's add some warnings in executions logs if user is using this option with non-ssh or not supported flow. Ideally we should have another methods somethings like configureCredentialsForModuleSource and then in that method add warnings and use exportSSHKey
</blockquote></details>

Done.
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 818 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B384M01a4z8VEHzQC:-N0Fjqkn0nzu9idNU5LI:b-u05b2x) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L818)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

I'm confused here why do we need to append USER_DIR_KEY? you can get absolute path by using Path.getAbsolutePath()
</blockquote></details>

not able to find another to get delegate path other than System.getProperty(USER_DIR_KEY)
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 829 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B3Rzp4lfIaRbF_DHR:-N0FkKsJ5We1oewmCG1c:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L829)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Let's add a comment here why we need this manipulation with file permission
</blockquote></details>

Done.
___
*[125-cd-nextgen/src/main/java/io/harness/cdng/manifest/yaml/storeConfig/moduleSource/ModuleSource.java, line 29 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B28BBASFnz0BXQhQu:-N0Fjh5g4vwZrIOx3Q67:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/125-cd-nextgen/src/main/java/io/harness/cdng/manifest/yaml/storeConfig/moduleSource/ModuleSource.java#L29)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

as discussed this should be runtime input
</blockquote></details>

Done.

**Review by namantalaycha10 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)* status: 28 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java, line 820 at r2](https://reviewable.io/reviews/harness/harness-core/32363#-N0B2oKJ5iUIztx-fbln:-N0FmWOy6bFcXxQq3NoJ:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/b221a7f912113fd9c710d69723d4f17fa917394c/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/TerraformBaseHelperImpl.java#L820)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: sshKey should be const, and we can name it something like ssh.key
</blockquote></details>

Done

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 12 of 18 files at r3, 13 of 15 files at r4, 3 of 3 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)* status: 1 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformApplyStep.java, line 149 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GA6Zt6SWohTHovFhP:-N0GA6Zt6SWohTHovFhQ:b9xahqd) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformApplyStep.java#L149)):*
> ```Java
>         .terraformCommandUnit(TerraformCommandUnit.Apply)
>         .entityId(entityId)
>         .tfModuleSourceInheritSSH(cdFeatureFlagHelper.isEnabled(
> ```

nit: we can move this check before builder in temp variable. it's a bit hard to follow, something like:
boolean tfModuleSourceInheritSSH = false;
if (ffEnabled) {
  tfModuleSourceInheritSSH = <all this logic>
}
or even can be moved to terraform step helper
___
*[930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java, line 371 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GHgRkDMmuI6Nnv0rE:-N0GHgRn-bDFhWwzrdgo:b-hnq92y) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L371)):*
> ```Java
>     assertThat(taskNGParameters.getEnvironmentVariables().get(GIT_SSH_COMMAND))
>         .contains("ssh -o StrictHostKeyChecking=no -o BatchMode=yes -o PasswordAuthentication=no -i");
>     assertThat(taskNGParameters.getEnvironmentVariables().get(GIT_SSH_COMMAND))
> ```

can we add a check if file exists as well and delete it in finally block try { ... } finally { FileIo.deleteFileIfExists(new File("./terraform-working-dir/entityId/.ssh/ssh.key"))
___
*[930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java, line 413 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GHtu_7D9y2RZxg_fu:-N0GHtu_7D9y2RZxg_fv:b-n2zcw7) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L413)):*
> ```Java
>     assertThat(taskNGParameters.getEnvironmentVariables().get(GIT_SSH_COMMAND)).contains("ssh.key");
> 
>     FileIo.deleteFileIfExists(Paths.get(System.getProperty(USER_DIR_KEY), "ssh.key").toString());
> ```

I'm not sure why it should be in USER_DIR_KEY, we changed this logic, as well let's add a check if file exists and add this delete file in finally block
___
*[930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java, line 497 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GI8LH7zPPTyXw7D6w:-N0GI8LH7zPPTyXw7D6x:buo9nmt) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L497)):*
> ```Java
>             WARN);
>     assertThat(taskNGParameters.getEnvironmentVariables().size()).isEqualTo(0);
>     FileIo.deleteFileIfExists(Paths.get(System.getProperty(USER_DIR_KEY), "ssh.key").toString());
> ```

Same here
___
*[930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java, line 525 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GI9aK2dwOeYOpo5we:-N0GI9aK2dwOeYOpo5wf:buo9nmt) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L525)):*
> ```Java
>             WARN);
>     assertThat(taskNGParameters.getEnvironmentVariables().size()).isEqualTo(0);
>     FileIo.deleteFileIfExists(Paths.get(System.getProperty(USER_DIR_KEY), "ssh.key").toString());
> ```

Same here
___
*[970-api-services-beans/src/main/java/io/harness/provision/TerraformConstants.java, line 67 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GIOh5Dy5SJuAlm911:-N0GIOh5Dy5SJuAlm912:b-idx9bi) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/970-api-services-beans/src/main/java/io/harness/provision/TerraformConstants.java#L67)):*
> ```Java
>       " -o StrictHostKeyChecking=no -o BatchMode=yes -o PasswordAuthentication=no -i ";
>   public static final String SSH_KEY_DIR = ".ssh";
>   public static final String USE_CONNECTOR_CREDENTIALS = "useConnectorCredentials";
> ```

is this used anywhere?

**Review by namantalaycha10 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)* status: 34 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java` line 371 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GHgRkDMmuI6Nnv0rE:-N0_fJ0EEJ9pLxyLMbBO:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L371)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

can we add a check if file exists as well and delete it in finally block try { ... } finally { FileIo.deleteFileIfExists(new File("./terraform-working-dir/entityId/.ssh/ssh.key"))
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java` line 413 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GHtu_7D9y2RZxg_fu:-N0_fJp54Lvk5hi5S122:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L413)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

I'm not sure why it should be in USER_DIR_KEY, we changed this logic, as well let's add a check if file exists and add this delete file in finally block
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java` line 497 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GI8LH7zPPTyXw7D6w:-N0_fKLp55sn1HJgyxPb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L497)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java` line 525 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GI9aK2dwOeYOpo5we:-N0_fKkf3c9dUOoo7DLB:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/TerraformBaseHelperImplTest.java#L525)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Done.
___
*[`970-api-services-beans/src/main/java/io/harness/provision/TerraformConstants.java` line 67 at r6](https://reviewable.io/reviews/harness/harness-core/32363#-N0GIOh5Dy5SJuAlm911:-N0_fLWN06TV_WDGV23H:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/52dca7377f36ffba10da1a2b4cb8761e6bc8b6d2/970-api-services-beans/src/main/java/io/harness/provision/TerraformConstants.java#L67)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

is this used anywhere?
</blockquote></details>

Done.

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 1]
```diff
@@ -318,6 +318,7 @@ public enum FeatureName {
   HIDE_ABORT,
   CUSTOM_ARTIFACT_NG,
   APPLICATION_DROPDOWN_MULTISELECT,
+  TF_MODULE_SOURCE_INHERIT_SSH,
```
Comment (abhinav-harness) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:321]: add at end

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/terraform/TerraformTaskNGParameters.java [hunk 2]
```diff
@@ -57,6 +57,7 @@
   @Expression(DISALLOW_SECRETS) List<String> targets;
   @Expression(ALLOW_SECRETS) Map<String, String> environmentVariables;
   boolean saveTerraformStateJson;
+  boolean tfModuleSourceInheritSSH;
```
Comment (abhinav-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/terraform/TerraformTaskNGParameters.java:60]: we need to handle case when it is null, also i think this should be Boolean

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java (hunk 1 above)
Comment (namantalaycha10) [reply 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java]: done

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 1 of 13 files at r1, 3 of 10 files at r2, 11 of 18 files at r3, 4 of 15 files at r4, 5 of 12 files at r8, 18 of 20 files at r9, 2 of 4 files at r11, 3 of 3 files at r12, 2 of 2 files at r13, 1 of 1 files at r14, 2 of 2 files at r15, 1 of 1 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32363)* status: 1 change requests. 5. unresolved discussions. 0 of 1 approvals obtained.

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/terraform/TerraformTaskNGParameters.java (hunk 2 above)
Comment (namantalaycha10) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/terraform/TerraformTaskNGParameters.java:60]: yes @abhinav-harness I handled the false case
